src/compiler/parser.rs

```diff
@@ -588,20 +536,30 @@ impl ParserType {
         result
     }
 
-    fn look_ahead<TReturn, TCallback: FnMut() -> Option<TReturn>>(
+    fn look_ahead<TReturn, TCallback: FnOnce() -> Option<TReturn>>(
         &self,
         callback: TCallback,
     ) -> Option<TReturn> {
         self.speculation_helper(callback, SpeculationKind::Lookahead)
     }
 
-    fn try_parse<TReturn, TCallback: FnMut() -> Option<TReturn>>(
+    fn look_ahead_bool<TCallback: FnOnce() -> bool>(&self, callback: TCallback) -> bool {
```

Comment:
The Typescript version of `lookAhead()`/`tryParse()` just looks at the falsiness of the callback return value, but in Rust sometimes that would correspond to an `Option` and sometimes to a `bool` so provide these `*_bool()` variants to make calling them more ergonomic in that case